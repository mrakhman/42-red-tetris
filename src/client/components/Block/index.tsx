import { StyleSheet, View } from 'react-native';
import * as React from 'react';

import { cellState } from '/config/constants';

const styles = StyleSheet.create({
  outer: {
    alignItems: 'center',
    borderColor: '#879372',
    borderWidth: 1,
    height: 20,
    justifyContent: 'center',
    width: 20,
  },
  outerSpectrum: {
    alignItems: 'center',
    borderColor: '#879372',
    borderWidth: 1,
    height: 6,
    justifyContent: 'center',
    width: 6,
  },
  inner: {
    height: 12,
    width: 12,
    backgroundColor: '#879372',
  },
  innerSpectrum: {
    height: 2,
    width: 2,
    backgroundColor: '#879372',
  },
});

const Block = ({ value, isSpectrum }: { value: CellStateType | number, isSpectrum?: boolean }): JSX.Element => {
  return (
    isSpectrum
      ?
      <View style={[styles.outerSpectrum, value === cellState.OCCUPIED ? { borderColor: 'black' } : (value === cellState.BLOCKED ? { borderColor: 'black'} : undefined)]}>
        <View style={[styles.innerSpectrum, value === cellState.OCCUPIED ? { backgroundColor: 'black' } : (value === cellState.BLOCKED ? { backgroundColor: 'transparent'} : undefined)]} />
      </View>
      :
      <View style={[styles.outer, value === cellState.OCCUPIED ? { borderColor: 'black' } : (value === cellState.BLOCKED ? { borderColor: 'black'} : undefined)]}>
        <View style={[styles.inner, value === cellState.OCCUPIED ? { backgroundColor: 'black' } : (value === cellState.BLOCKED ? { backgroundColor: 'transparent'} : undefined)]} />
      </View>
  );
};

export default Block;
