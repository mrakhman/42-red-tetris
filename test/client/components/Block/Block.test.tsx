import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Block from '/client/components/Block';

describe('Snapshots', () => {
  it('should render Block with cellState.FREE', () => {
    const props = {
      value: 0,
    };
    const wrapper = shallow(<Block {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render Block with cellState.OCCUPIED', () => {
    const props = {
      value: 1,
    };
    const wrapper = shallow(<Block {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render Block with cellState.BLOCKED', () => {
    const props = {
      value: 2,
    };
    const wrapper = shallow(<Block {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
