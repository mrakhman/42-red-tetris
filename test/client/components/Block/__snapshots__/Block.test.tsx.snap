// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render Block with cellState.BLOCKED 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#879372",
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "borderColor": "black",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#879372",
          "height": 12,
          "width": 12,
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  />
</View>
`;

exports[`Snapshots should render Block with cellState.FREE 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#879372",
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#879372",
          "height": 12,
          "width": 12,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Snapshots should render Block with cellState.OCCUPIED 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "#879372",
        "borderWidth": 1,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "borderColor": "black",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#879372",
          "height": 12,
          "width": 12,
        },
        Object {
          "backgroundColor": "black",
        },
      ]
    }
  />
</View>
`;
